### PR TITLE
Add flags for hostname and ports

### PR DESCRIPTION
**Pull request by fortuna**
This will allow advanced users to easily specify a domain name for their server, and what ports to use.

Example use:
```
sudo bash -c "$(wget -qO- https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh)" \
    install_server.sh --hostname=outline.mydomain.com --keys-port=443 --api-port=8443
```

### CODE REVIEW

File: src/server_manager/install_scripts/install_server.sh [hunk 1]
```diff
@@ -406,5 +433,54 @@ ${FIREWALL_STATUS}
...
+        display_usage
+        exit 1
+        ;;
+      *) # preserve positional arguments
+        PARAMS="$PARAMS '$1'"
+        ;;
+    esac
+    shift
+  done
+  if (( FLAGS_PORT_FOR_API == FLAGS_PORT_FOR_KEYS )); then
```
Comment (alalamav) [inline src/server_manager/install_scripts/install_server.sh]: Should we check that these are in the valid range (0-65535)?

File: src/server_manager/install_scripts/install_server.sh [hunk 2]
```diff
@@ -29,11 +27,23 @@
...
 
 # Requires curl and docker to be installed
 
 set -euo pipefail
 
+function display_usage() {
+  echo Usage: install_server.sh [--hostname <hostname>] [--port-for-api <port>] [--port-for-keys <port>]
+  echo
+  echo  --hostname        The hostname to be used to access the management API and access keys
+  echo  --port-for-api    The port number for the management API
```
Comment (alalamav) [inline src/server_manager/install_scripts/install_server.sh]: Consider renaming to `--api-port`, `--keys-port`. Same with the `FLAGS` variables.

File: src/server_manager/install_scripts/install_server.sh [hunk 3]
```diff
@@ -406,5 +433,54 @@ ${FIREWALL_STATUS}
...
+    log_error "Missing $1 value"
+    display_usage
+    exit 1
+  fi
+  echo $2
+}
+
+function parse_flags() {
+  local PARAMS=""
+  while [[ "$#" > 0 && ! "$1" == "--" ]]; do
```
Comment (alalamav) [inline src/server_manager/install_scripts/install_server.sh]: Should we use `getopts`? We wouldn't need to call eval, which could open a vulnerability.

File: src/server_manager/install_scripts/install_server.sh (hunk 1 above)
Comment (fortuna) [reply src/server_manager/install_scripts/install_server.sh]: Done

File: src/server_manager/install_scripts/install_server.sh [hunk 4]
```diff
@@ -29,11 +27,25 @@
...
+# SB_PUBLIC_IP: Use the --hostname flag instead
+# SB_API_PORT: Use the --apit-port flag instead
 
 # Requires curl and docker to be installed
 
 set -euo pipefail
 
+function display_usage() {
+  cat <<EOF
+Usage: install_server.sh [--hostname <hostname>] [--apit-port <port>] [--keys-port <port>]
```
Comment (alalamav) [inline src/server_manager/install_scripts/install_server.sh]: nit: typo --api-port

File: src/server_manager/install_scripts/install_server.sh [hunk 5]
```diff
@@ -29,11 +27,25 @@
 #     only by do_install_server.sh.
 # WATCHTOWER_REFRESH_SECONDS: refresh interval in seconds to check for updates,
 #     defaults to 3600.
+#
+# Deprecated:
+# SB_PUBLIC_IP: Use the --hostname flag instead
+# SB_API_PORT: Use the --apit-port flag instead
```
Comment (alalamav) [inline src/server_manager/install_scripts/install_server.sh]: nit: typo --api-port

**Comment by NightcoreIsla** [comment]
if use "--keys-port" flag
install bash will stuck at
Waiting for Outline server to be healthy...

**Pull request merged**